examples/long-short.py

```diff
@@ -11,328 +12,275 @@
-    # Format the allStocks variable for use in the class.
-    self.allStocks = []
-    for stock in stockUniverse:
-      self.allStocks.append([stock, 0])
-
-    self.long = []
-    self.short = []
-    self.qShort = None
-    self.qLong = None
-    self.adjustedQLong = None
-    self.adjustedQShort = None
-    self.blacklist = set()
-    self.longAmount = 0
-    self.shortAmount = 0
-    self.timeToClose = None
-
-  def run(self):
-    # First, cancel any existing orders so they don't impact our buying power.
-    orders = self.alpaca.list_orders(status="open")
-    for order in orders:
-      self.alpaca.cancel_order(order.id)
+    def __init__(self):
+        self.alpaca = tradeapi.REST(API_KEY, API_SECRET, URL(APCA_API_BASE_URL), 'v2')
+
+        stock_universe = ['DOMO', 'TLRY', 'SQ', 'MRO', 'AAPL', 'GM', 'SNAP', 'SHOP',
+                          'SPLK', 'BA', 'AMZN', 'SUI', 'SUN', 'TSLA', 'CGC', 'SPWR',
+                          'NIO', 'CAT', 'MSFT', 'PANW', 'OKTA', 'TWTR', 'TM', 'RTN',
+                          'ATVI', 'GS', 'BAC', 'MS', 'TWLO', 'QCOM', ]
+        # Format the allStocks variable for use in the class.
+        self.all_stocks = []
+        for stock in stock_universe:
+            self.all_stocks.append([stock, 0])
+
+        self.long = []
+        self.short = []
+        self.q_short = None
+        self.q_long = None
+        self.adjusted_q_long = None
+        self.adjusted_q_short = None
+        self.blacklist = set()
+        self.long_amount = 0
+        self.short_amount = 0
+        self.time_to_close = None
+
+    def run(self):
+        # First, cancel any existing orders so they don't impact our buying power.
+        orders = self.alpaca.list_orders(status="open")
+        for order in orders:
+            self.alpaca.cancel_order(order.id)
+
+        # Wait for market to open.
+        print("Waiting for market to open...")
+        self.await_market_open()
+        print("Market opened.")
+
+        # Rebalance the portfolio every minute, making necessary trades.
+        while True:
+
+            # Figure out when the market will close so we can prepare to sell beforehand.
+            clock = self.alpaca.get_clock()
+            closing_time = clock.next_close.replace(tzinfo=datetime.timezone.utc).timestamp()
+            curr_time = clock.timestamp.replace(tzinfo=datetime.timezone.utc).timestamp()
+            self.time_to_close = closing_time - curr_time
+
+            if self.time_to_close < (60 * 15):
+                # Close all positions when 15 minutes til market close.
+                print("Market closing soon.  Closing positions.")
+
+                positions = self.alpaca.list_positions()
+                for position in positions:
+                    if position.side == 'long':
+                        order_side = 'sell'
+                    else:
+                        order_side = 'buy'
+                    qty = abs(int(float(position.qty)))
```

Comment:
Can you please explain why we need the float conversion here?

Reply:
There are a lot more similar double `int(float(...))` conversions throughout the file.